internal/graph/benchmark_test.go

```diff
@@ -0,0 +1,146 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package graph
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"testing"
+
+	"github.com/opencontainers/go-digest"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"oras.land/oras-go/v2/content"
+	"oras.land/oras-go/v2/internal/cas"
+)
+
+func testSetUp(ctx context.Context) ([]ocispec.Descriptor, content.Fetcher) {
+	// set up for the test, prepare the test content and
+	// return the Fetcher
+	testFetcher := cas.NewMemory()
+
+	var blobs [][]byte
+	var descriptors []ocispec.Descriptor
+	appendBlob := func(mediaType string, blob []byte) ocispec.Descriptor {
+		blobs = append(blobs, blob)
+		descriptors = append(descriptors, ocispec.Descriptor{
+			MediaType: mediaType,
+			Digest:    digest.FromBytes(blob),
+			Size:      int64(len(blob)),
+		})
+		return descriptors[len(descriptors)-1]
+	}
+	generateManifest := func(layers ...ocispec.Descriptor) ocispec.Descriptor {
+		manifest := ocispec.Manifest{
+			Layers: layers,
+		}
+		manifestJSON, err := json.Marshal(manifest)
+		if err != nil {
+			panic(err)
+		}
+		return appendBlob(ocispec.MediaTypeImageManifest, manifestJSON)
+	}
+	generateIndex := func(manifests ...ocispec.Descriptor) ocispec.Descriptor {
+		index := ocispec.Index{
+			Manifests: manifests,
+		}
+		indexJSON, err := json.Marshal(index)
+		if err != nil {
+			panic(err)
+		}
+		return appendBlob(ocispec.MediaTypeImageIndex, indexJSON)
+	}
+	descE := appendBlob("layer node E", []byte("Node E is a layer")) // blobs[0], layer "E"
+	descF := appendBlob("layer node F", []byte("Node F is a layer")) // blobs[1], layer "F"
+	descB := generateManifest(descriptors[0:1]...)                   // blobs[2], manifest "B"
+	descC := generateManifest(descriptors[0:2]...)                   // blobs[3], manifest "C"
+	descD := generateManifest(descriptors[1:2]...)                   // blobs[4], manifest "D"
+	descA := generateIndex(descriptors[2:5]...)                      // blobs[5], index "A"
+	testFetcher.Push(ctx, descA, bytes.NewReader(blobs[5]))
+	testFetcher.Push(ctx, descB, bytes.NewReader(blobs[2]))
+	testFetcher.Push(ctx, descC, bytes.NewReader(blobs[3]))
+	testFetcher.Push(ctx, descD, bytes.NewReader(blobs[4]))
+	testFetcher.Push(ctx, descE, bytes.NewReader(blobs[0]))
+	testFetcher.Push(ctx, descF, bytes.NewReader(blobs[1]))
```

Comment:
nit: why not doing this using a loop?
BTW, we won't commit this file eventually, right?

Reply:
Good point. Maybe we will let `DeletableMemory` to replace `Memory`, and then we'll remove this file and others altogether.

Reply:
If I write this in a loop, then I'll have to put `descA/B/C/D/E/F` into an array of certain order, and I think I'll just save that effort and use copy and paste.

Reply:
I see. But I think the order of push does not matter for memory store.

Reply:
Removed the benchmark test file.